src/app/@ansyn/ansyn/app-providers/overlay-source-providers/planet/planet-source-provider.ts

```diff
@@ -97,12 +97,16 @@ export class PlanetSourceProvider extends BaseOverlaySourceProvider {
 			const configFilters = [];
 			const preFilter = { type: 'OrFilter', config: configFilters };
 			fetchParams.dataInputFilters.forEach((aFilter: DataInputFilterValue) => {
-				configFilters.push({
-					type: 'AndFilter', config: [
-						{ type: 'StringInFilter', field_name: 'item_type', config: [aFilter.sensorType] },
-						{ type: 'StringInFilter', field_name: 'satellite_id', config: [aFilter.sensorName] }
-					]
-				});
+				if (Boolean(aFilter.sensorName)) {
+					configFilters.push({
+						type: 'AndFilter', config: [
+							{ type: 'StringInFilter', field_name: 'item_type', config: [aFilter.sensorType] },
+							{ type: 'StringInFilter', field_name: 'satellite_id', config: [aFilter.sensorName] }
+						]
+					});
+				} else {
+					configFilters.push({ type: 'StringInFilter', field_name: 'item_type', config: [aFilter.sensorType] });
```

Comment:
How do you feel about the following suggestion? Instead of duplicating the sensorType filtering code, we can do the following:
```typescript
const conf = [ { type: 'StringInFilter', field_name: 'item_type', config: [aFilter.sensorType] } ]
if (Boolean(aFilter.sensorName)) {
    config.push({ type: 'StringInFilter', field_name: 'satellite_id', config: [aFilter.sensorName] })
}
configFilters.push({ type: 'AndFilter', conf })
```